modules/imgcodecs/include/opencv2/imgcodecs.hpp

```diff
@@ -227,6 +227,19 @@ Currently, the following file formats are supported:
 */
 CV_EXPORTS_W Mat imread( const String& filename, int flags = IMREAD_COLOR );
 
+
+/** @brief Loads an image from a file.
+
+This is an overloaded member function, provided for convenience. It differs from the above function only in what argument(s) it accepts and the return value.
+@param filename Name of file to be loaded.
+@param img object in which the image will be loaded.
+@param flags Flag that can take values of cv::ImreadModes
+@note
+The image passing trought the img parameter can be pre-allocated. The memory is reused if the shape and the type match with the load image.
+ */
+CV_EXPORTS_W void imread( const String& filename, Mat & img, int flags = IMREAD_COLOR );
```

Comment:
> `CV_EXPORTS_W void imread`

There are 2 options to avoid conflicts and ambiguity:
1. imread => imreadToBuffer
2. CV_EXPORTS_W => CV_EXPORTS_AS(imreadToBuffer)

Reply:
I don't have conflicts in Python binding using just `OuputArray`:
```python
import cv2 as cv

img = cv.imread("img1.png")
cv.imread("img2.png, img)
```

Reply:
it's all the same to me. how do we want to proceed?